src/main/java/graphql/normalized/NormalizedQueryTreeFactory.java

```diff
@@ -52,7 +53,18 @@ private NormalizedQueryTree createNormalizedQueryImpl(GraphQLSchema graphQLSchem
                 .variables(variables)
                 .build();
 
-        CollectFieldResult topLevelFields = fieldCollector.collectFromOperation(parameters, operationDefinition, graphQLSchema.getQueryType());
+        GraphQLObjectType rootType;
+        if (operationDefinition.getOperation() == OperationDefinition.Operation.QUERY) {
+            rootType = graphQLSchema.getQueryType();
+        } else if (operationDefinition.getOperation() == OperationDefinition.Operation.MUTATION) {
+            rootType = graphQLSchema.getMutationType();
+        } else if (operationDefinition.getOperation() == OperationDefinition.Operation.SUBSCRIPTION) {
+            rootType = graphQLSchema.getSubscriptionType();
+        } else {
+            throw new IllegalStateException("Operation type is not valid.");
```

Comment:
We use normally `Assert.assertShouldNeverHappen` here.

Reply:
Hello Andreas, 
`Assert.assertShouldNeverHappen` seems to not work here as compiler complains that `rootType` is not initialized. And it looks like a good compile time check.
Can we commit this (as original author seems to be not available) as is? And replace this exception with something like
```
// Don't use Assert.assertShouldNeverHappen to make sure compiler does not force
// rootType above to be initialized. That way if code structure changes, compiler will
// catch that. Otherwise, we'd get NullPointerException at runtime.
throw new AssertException(
        String.format(
                "Internal error: should never happen: Operation type '%s' is not valid.",
                operationDefinition.getOperation()));
```
in a follow up PR?

Reply:
@andimarek thoughts?

Reply:
+1 to commit, I'm hitting this issue as well. How about using a return statement to get rid of the initialization issue:
```java
return Assert.assertShouldNeverHappen("Operation type is not valid.");
```